### PR TITLE
[fundamental] Check enforcer should check against merge_base

### PR DESCRIPTION
**Pull request by crazygao**
# Description

Check enforcer should check against merge_base
move enforcer to new folder, and avoid trigger too many code for check_enforcer changes.

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: scripts/check_enforcer/check_enforcer.py [hunk 1]
```diff
@@ -209,8 +209,10 @@ def run_checks():
         merge_commit = (
             subprocess.check_output(["git", "log", "-1"]).decode("utf-8").split("\n")
         )
+        if snippet_debug == 1:
```
Comment (wangchao1230) [inline scripts/check_enforcer/check_enforcer.py:212]: when will this be 0?

File: .github/workflows/check_enforcer.yml [hunk 2]
```diff
@@ -12,7 +12,8 @@ jobs:
       contents: read
     steps:
       - uses: actions/checkout@v4
-      - run: git fetch origin main
+        with:
```
Comment (wangchao1230) [inline .github/workflows/check_enforcer.yml:15]: how will we test this change?

File: scripts/check_enforcer/check_enforcer.py (hunk 1 above)
Comment (crazygao) [reply scripts/check_enforcer/check_enforcer.py:212]: Just open this debug flag. If we don't want output and just get check enforcer output.
We can set this flag at the very top of the snippet

File: .github/workflows/check_enforcer.yml (hunk 2 above)
Comment (crazygao) [reply .github/workflows/check_enforcer.yml:15]: This branch is out of date. but check enforcer is passed.

File: scripts/check_enforcer/check_enforcer.py (hunk 1 above)
Comment (crazygao) [reply scripts/check_enforcer/check_enforcer.py:212]: But this is a good point, I'll make a switch here and let the yaml could control this snippet debug flag.

**Review by wangchao1230 — APPROVED** [review]

File: scripts/check_enforcer/check_enforcer.py [hunk 3]
```diff
@@ -26,7 +26,7 @@
 
 # Define variables
 github_repository = "microsoft/promptflow"
-snippet_debug = 1  # Write debug info to console.
+snippet_debug = os.getenv("SNIPPET_DEBUG", 0)
```
Comment (zhengfeiwang) [inline scripts/check_enforcer/check_enforcer.py:29]: when will this environ be set?

**Review by zhengfeiwang — APPROVED** [review]

**Pull request merged**